native/explorer/src/encoding.rs

```diff
@@ -138,10 +138,11 @@ fn naive_datetime_struct_keys(env: Env) -> [NIF_TERM; 9] {
 }
 
 #[inline]
-fn time_unit_to_factor(time_unit: TimeUnit) -> i64 {
+fn time_unit_to_factor(time_unit: TimeUnit) -> f64 {
```

Comment:
@xaviRodri I would suggest use this function directly to convert the value to microseconds, and you only use  floats for the case of nanoseconds. With this we should only have small imprecision with nanoseconds.

something like:

```rust
fn time_to_microseconds(v: i64, time_unit: TimeUnit) -> i64 {
  match time_unit {
      TimeUnit::Milliseconds => v * 1000,
      TimeUnit::Microseconds => v,
      TimeUnit::Nanoseconds => (v as f64 * 0.001) as i64,
      _ => unreachable!(),
  }
}
```

WDYT?

Reply:
Hey!!
@philss thanks a lot for the suggestion! 👏 
Makes more sense so we minimise the imprecision as much as possible 👍 
I will try then to implement some `.parquet` tests with this and push again for another review!